[integration] added script for multiVO certification

marianne013 opened the pull request:
BEGINRELEASENOTES

*tests
NEW: Added a script intended for multi-VO certification that actually uses two different VOs. If approved would need a trello card as well.


ENDRELEASENOTES

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+testdir=`echo $PWD`
```
I might be missing something but:
```suggestion
testdir=$PWD
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+
+EXEFILE="${testdir}/multiVOexe.sh"
+cat > ${EXEFILE} <<EOF
+echo $DIRACSITE
+ghostname=`hostname --long 2>&1`
+gipname=`hostname --ip-address 2>&1`
+echo $ghostname "has address" $gipname
```
Does this do what you intend? Or did you mean:
```suggestion
echo \$DIRACSITE
ghostname=\$(hostname --long 2>&1)
gipname=\$(hostname --ip-address 2>&1)
echo \$ghostname "has address" \$gipname
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+cat /etc/redhat-release
+EOF
+
+MYDATE=`date +%s`
```
https://github.com/koalaman/shellcheck/wiki/SC2006
```suggestion
MYDATE=$(date +%s)
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+dirac-wms-job-status -f dteam_jobid.log
+
+echo -e "\nUploading file as a dteam_user"
+dirac-dms-add-file /dteam/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk
+if [[ "${?}" -ne 0 ]]; then
```
```suggestion
if dirac-dms-add-file "/dteam/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt" UKI-LT2-IC-HEP-disk; then
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+export X509_USER_PROXY=${testdir}/dteam.proxy
+echo -e "\nChanging VO to dteam."
+echo "dirac-proxy-init -g dteam_user --out ${testdir}/dteam.proxy"
+dirac-proxy-init -g dteam_user --out ${testdir}/dteam.proxy
+if [[ "${?}" -ne 0 ]]; then
```
```suggestion
if dirac-proxy-init -g dteam_user --out "${testdir}/dteam.proxy"; then
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+dirac-wms-job-status -f gridpp_jobid.log
+
+echo -e "\nUploading file as a gridpp_user"
+dirac-dms-add-file /gridpp/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk
+if [[ "${?}" -ne 0 ]]; then
```
```suggestion
if dirac-dms-add-file "/gridpp/diraccert/testfile.${MYDATE}.txt testfile.${MYDATE}.txt UKI-LT2-IC-HEP-disk"; then
```

chrisburr commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,82 @@
+# need to tell DIRAC beforehand where the user proxy will be
+export X509_USER_PROXY=${testdir}/gridpp.proxy
+echo "dirac-proxy-init -g gridpp_user --out ${testdir}/gridpp.proxy"
+dirac-proxy-init -g gridpp_user --out ${testdir}/gridpp.proxy
+if [[ "${?}" -ne 0 ]]; then
```
```suggestion
if dirac-proxy-init -g gridpp_user --out "${testdir}/gridpp.proxy"; then
```

marianne013 commented:
Sorry, I had a previous incarnation which worked, then did an update to make it "better", but currently jobs in the certification instance don't seem to be going anywhere, so I didn't really test it (Friday afternoon and all). I see if I can rig some tests against my own instance and submit an update.

marianne013 commented:
I think I addressed all the comments, apart from the if statement. I think that's a style issue and the current implementation is easier to read than shellcheck's. I've tested it on my local server as much as possible, but as I don't have a dteam VO enabled, I can't test everything.

andresailer commented on tests/System/client_multivo.sh:
```diff
@@ -0,0 +1,85 @@
+
+echo -e "\nSubmitting job as a gridpp user"
+dirac-wms-job-submit -f gridpp_jobid.log multiVO.jdl
+sleep 30
```
Add `echo "sleeping 30 s"` so people don't wonder why nothing happens?
